src/Instance/InstanceEdit/InstanceEdit.js

```diff
@@ -54,12 +55,17 @@ const InstanceEdit = ({
   const goBack = useGoBack(`/inventory/view/${instanceId}`);
 
   const onSuccess = useCallback((updatedInstance) => {
-    callout.sendCallout({
-      type: 'success',
-      message: <FormattedMessage
+    const successfullySavedMessage = (
+      <FormattedMessage
         id="ui-inventory.instance.successfullySaved"
         values={{ hrid: updatedInstance.hrid }}
-      />,
+      />
+    );
+    const successfullySavedSharedMessage = <FormattedMessage id="ui-inventory.instance.shared.successfulySaved" />;
+
+    callout.sendCallout({
+      type: 'success',
+      message: instance?.shared ? successfullySavedSharedMessage : successfullySavedMessage,
```

Comment:
This is a little simpler and avoids creating `successfullySavedSharedMessage` if it won't get used:
```
const message = instance?.shared ? <FormattedMessage shared message> : <FormattedMessage regular message>;
...
calloutSendCallout({ type: 'success', message }); 
```